### PR TITLE
Feature 1013/implement feedback request controllers integration tests

### LINKED ISSUE #1013: BE: Implement feedback request controllers and integration tests (M)
- [x] Implement save function in controller
- [x] Implement integration test for save under testing folder
- [x] Implement delete function in controller
- [x] Implement integration test for delete under testing folder
- [x]  Implement getById function in controller
- [x] Implement integration test for getById under testing folder
- [x] Implement getByCreatorId function in controller
- [x] Implement integration test for getByCreatorId under testing folder

### PR DESCRIPTION
**Pull request by pieperm**
To test this, run:
```
./gradlew test --tests FeedbackRequestControllerTest
```

This feature includes:
- Controller for feedback requests
- Implementations for `save`, `delete`, `getById`, `findByValue`
- Authorization for certain operations based on permissions
- Integration tests

Closes #1013

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    @Schema(description = "id of the person who is getting feedback requested on them", required = true)
+    private UUID requesteeId;
+
+    @Column(name = "templateId")
+    @NotNull
+    @TypeDef(type = DataType.STRING)
+    @Schema(description = "id of the template attached to request", required = true)
+    private UUID templateId;
+
+    @Column(name="sendDate")
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequest.java:46]: is this a nullable column ?

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestRepository.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Set;
+import java.util.UUID;
+
+@JdbcRepository(dialect = Dialect.POSTGRES)
+public interface FeedbackRequestRepository extends CrudRepository<FeedbackRequest, UUID> {
+
+    @Override
+    <S extends FeedbackRequest> S save(@Valid @NotNull @Nonnull S entity);
+
+    @Query(value = "SELECT * from feedback_requests WHERE creatorId = :creatorId ")
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestRepository.java:20]: is there a reason we defined custom query here ?

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                        .headers(headers -> headers.location(URI.create("/feedback_request" + savedFeedbackRequest.getId()))))
+                .subscribeOn(Schedulers.from(executorService));
+    }
+    /**
+     * Get feedback request by creator's ID
+     *
+     * @param creatorId {@link UUID} ID of member profile who created the feedback request
+     * @return {@link Set <FeedbackResponseDTO>} Set of feedback requests that were made by certain creator
+     */
+    @Get("/{?creatorId}")
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java:90]: do we need a get call for `requesteeId` as well ?

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequest.java (hunk 1 above)
Comment (pieperm) [reply server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequest.java:46]: No, it is not. I just added the `@NotNull` annotation to that column

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestRepository.java (hunk 2 above)
Comment (pieperm) [reply server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestRepository.java:20]: No, I have removed it and adjusted the call to `findByCreatorId` accordingly

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java (hunk 3 above)
Comment (pieperm) [reply server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java:90]: We can't think of a use case where we would only want to see feedback requests on a specific person (at least not by ID). We would want to see feedback requests filtered by the *name* of the requestee though.

**Review by moshirim — APPROVED** [review]

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
+package com.objectcomputing.checkins.services.feedback_request;
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_request/FeedbackRequestController.java:1]: will there be a use case where we will need `update` end point ? for example after sending an feedback request, I want to update the template ?

**Comment by mkimberlin** [comment]
I was with this PR until I got to HasFeedbackRequest. I don't get what this bit was meant to do and I don't think the related classes are connected to anything.

**Comment by pieperm** [comment]
@mkimberlin They were not related to anything. We started making that relation before we decided to split it into a separate issue (#1024). Since they are on that branch, I have removed `has_feedback_request` from this one

**Pull request closed without merging**